Sources/Controllers/MessagesViewController.swift

```diff
@@ -72,7 +72,37 @@ open class MessagesViewController: UIViewController {
         registerReusableViews()
         setupDelegates()
 
+		self.addMenuControllerObservers()
     }
+	
+	fileprivate func addMenuControllerObservers() {
+		NotificationCenter.default.addObserver(self, selector: #selector(MessagesViewController.menuControllerWillShow(aNotification:)), name: NSNotification.Name.UIMenuControllerWillShowMenu, object: nil)
+		NotificationCenter.default.addObserver(self, selector: #selector(MessagesViewController.menuControllerDidHide(aNotification:)), name: NSNotification.Name.UIMenuControllerDidHideMenu, object: nil)
+	}
+	
+	fileprivate func removeMenuControllerObservers() {
+		NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIMenuControllerWillShowMenu, object: nil)
+		NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIMenuControllerDidHideMenu, object: nil)
+	}
+	
+	fileprivate var targetedCell:UICollectionViewCell?
+	fileprivate var targetedFrame = CGRect.zero
```

Comment:
Can we keep properties at the top of the file like we're doing with the rest of them?